### ISSUE TITLE
Config issues

### ISSUE THREAD

**Issue by andrewzigerelli** [hadolint/hadolint#624]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Problem 
.hadolint.yaml appears to be ignored.
It appears to be loaded correctly, as it detects errors for malformed .yaml.

### Expected behavior
Ignored rules from .yaml file are actually ignored, in this example ignore DL3059.

### Actual behavior
DL3059 not ignored.

### Steps to reproduce the behavior

**Output of `hadolint Dockerfile` or
  `hadolint --config .hadolint.yaml`**

```bash 
Dockerfile:8 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
Dockerfile:8 SC1091 info: Not following: File not included in mock.
```

**Dockerfile** (if relevant)

```Dockerfile
FROM ubuntu:20.04

SHELL ["/bin/bash", "-l", "-c", "-o", "pipefail"]

WORKDIR /root/

RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
RUN source /root/.nvm/nvm.sh && \
nvm install node
```

**.hadolint.yaml**
```yaml
ignored:
  - DL3059
```

**Additional environment details (OS, stack version, etc.)**
stack 2.5.1
I cloned from the git repo and installed from source.

### Another related issue
The README says I can automatically use ~/.config/hadolint.yaml, but it looks like 4b949a832eacad7f0065d94027df1208f441fd97 doesn't check this directory (I don't know much Haskell though). I tried by deleting the hadolint.yaml in the same directory as the Dockerfile, and forcing ~/.config/hadolint.yaml to be malformed. It's not loaded unless I manually pass it with -c option.

**Comment by lorenzo** [comment]
you built hadolint from source? Asking since that commit is not in any release yet

**Comment by andrewzigerelli** [comment]
> you built hadolint from source? Asking since that commit is not in any release yet

yes, sorry, I included it in the details now.
